tests/unit/Core/Checkout/Document/Service/DocumentMergerTest.php

```diff
@@ -30,23 +31,110 @@
 #[CoversClass(DocumentMerger::class)]
 class DocumentMergerTest extends TestCase
 {
-    public function testMerge(): void
+    public const PDF_CONTENT = 'PDF content for testing';
+
+    public function testMergeOneDocument(): void
     {
-        $orderId = Uuid::randomHex();
+        $document = $this->createDocument(true);
 
-        $documentType = new DocumentTypeEntity();
-        $documentType->setId(Uuid::randomHex());
-        $documentType->setTechnicalName('invoice');
+        $fpdi = $this->createMock(Fpdi::class);
+        $fpdi->expects($this->never())->method('setSourceFile');
 
-        $document = new DocumentEntity();
-        $document->setId(Uuid::randomHex());
-        $document->setOrderId($orderId);
-        $document->setDocumentTypeId($documentType->getId());
-        $document->setDocumentType($documentType);
-        $document->setStatic(false);
-        $document->setConfig([]);
+        $documentGenerator = $this->createMock(DocumentGenerator::class);
+        $documentGenerator->expects($this->never())->method('generate');
+
+        $mediaService = $this->createMock(MediaService::class);
+        $mediaService->method('loadFile')->willReturn(self::PDF_CONTENT);
+
+        $documentRepository = $this->createMock(EntityRepository::class);
+        $documentRepository->expects($this->once())->method('search')->willReturn(
+            new EntitySearchResult(
+                'document',
+                1,
+                new DocumentCollection([$document]),
+                null,
+                new Criteria(),
+                Context::createDefaultContext(),
+            )
+        );
+
+        $documentMerger = new DocumentMerger(
+            $documentRepository,
+            $mediaService,
+            $documentGenerator,
+            $this->createMock(Fpdi::class),
+        );
+
+        $result = $documentMerger->merge(
+            [$document->getId()],
+            Context::createDefaultContext()
+        );
+
+        static::assertNotNull($result);
+        static::assertEquals('pdf', $result->getFileExtension());
```

Comment:
```suggestion
        static::assertSame('pdf', $result->getFileExtension());
```

please use `assertSame` where possible instead, as it is type safe. please also fix all other occurrences